### PR TITLE
Allow uploading package when metadata and filesystem are out of sync

### PR DESCRIPTION
**Pull request by chefsalim**
This change allows uploading packages to the depot in a scenario where the package metadata and the on-disk archive gets out of sync. This can happen in a development or acceptance environment, and this change allows a simple 'hab pkg upload' to recover the environment. There is also a bug fix where the package visibility was not getting added to an OriginPackageGet call.

Signed-off-by: Salim Alam <salam@chef.io>

tenor-163584967

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/server.rs [hunk 1]
```diff
@@ -743,19 +744,29 @@ fn upload_package(req: &mut Request) -> IronResult<Response> {
 
     let mut ident_req = OriginPackageGet::new();
     ident_req.set_ident(ident.clone());
+    ident_req.set_visibilities(visibility_for_optional_session(
+        req,
+        session_id,
+        &ident.get_origin(),
+    ));
```
Comment (raskchanky) [inline components/builder-depot/src/server.rs]: Good grief - how did this ever work?  Nice catch, regardless.

**Comment by raskchanky** [comment]
tenor-218810749

**Comment by elliott-davis** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/301595168).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/301595168).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**